parser.ts

```diff
@@ -72,6 +77,19 @@ class Synchronize extends Error {
   }
 }
 
+// While parsing instructions, certain situations indicate what must be a
+// final instruction. For instance, `if <condition>` without a trailing "then"
+// may not be followed by any instructions. In those cases, we throw an
+// exception to stop parsing of instructions and force going to the next
+// line.
+@errorType('StopInstrs')
+class StopInstrs extends Error {
```

Comment:
I don't think this is necessary anymore, if I'm going to drop the "then on next line" feature.